core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -548,6 +548,7 @@ object JarCreationTest extends Logging {
     if (result.nonEmpty) {
       throw new Exception("Could not load user class from jar:\n" + result(0))
     }
+    sc.stop()
```

Comment:
These are run in separate processes, no?  Shouldn't the shutdown hook registered by SparkContext itself take care of calling `stop` already?

 I'm not sure where the logs for these child processes end up, but you should be able to check if the "Successfully stopped SparkContext" message shows up.